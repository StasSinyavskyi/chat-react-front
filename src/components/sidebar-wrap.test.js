/* eslint-env jest */
import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';
import SidebarWrap from './sidebar-wrap';

jest.mock('./sidebar', () => () => 'Sidebar');

// if we have several test we can join them in describe
describe('<SidebarWrap />', () => {
  const testParam = (
    <SidebarWrap
      createChat={() => 'createChat'}
      chats={{ active: {}, my: [], all: [] }}
      activeChats={{ }}
      isConnected
    />
  );

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(testParam, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('render correctly', () => {
    const tree = renderer
      .create(testParam)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
