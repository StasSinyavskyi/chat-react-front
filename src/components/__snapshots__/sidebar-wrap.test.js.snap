// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarWrap /> render correctly 1`] = `
<div>
  <div
    className="MuiDrawer-docked-39"
  >
    <div
      className="MuiPaper-root-50 MuiPaper-shadow0-52 MuiDrawer-paper-40 MuiDrawer-paperAnchorLeft-41 MuiDrawer-paperAnchorDockedLeft-45"
    >
      Sidebar
    </div>
  </div>
</div>
`;
